### DISCUSSION TITLE
Thoughts about learned vs. computed virial/stress tensor and small system periodic artifacts for liquids

### DISCUSSION THREAD

**Discussion by asedova** [deepmodeling/deepmd-kit#2277] [Q&A]
It's well known that getting the virial/pressure right for liquid systems when using small box sizes, like used in DFT calculations, and with periodic boundaries, will result in errors from the liquid being more or less a "liquid crystal" and not a true liquid. DFT codes like VASP and CP2K calculate the stress tensor/virial using a derivation based on the assumption of periodic solids. For simulating liquids, there will be box size artifacts because of this.

So, I am wondering, when learning the virial (or stress tensor) in a machine learning potential, and then deploying on large systems for inference, it would seem that these small size/periodic system artifacts for liquids would propagate to the larger systems. An alternate strategy would be, I guess, to not learn the virial, but let the MD engine (for instance LAMMPS), calculate the stress tensor/pressure/virial using a force-based approach (and using the general method that allows this to be defined for many-body potentials).

Does anyone have any insights on this?

**Comment by njzjz** [comment]
> calculate the stress tensor/pressure/virial using a force-based approach (and using the general method that allows this to be defined for many-body potentials).

DeePMD-kit already uses the force-based approach to calculate and fit virials. You can read the Appendix of [the DeePMD-kit paper](https://www.sciencedirect.com/science/article/pii/S0010465518300882).

**Reply by asedova** [reply]
Ok, so is DeePMD-kit passing the virial to LAMMPS during the MD inference even if the virial has not been learned?

**Reply by njzjz** [reply]
Yes.
